src/decoder.rs

```diff
@@ -3,38 +3,133 @@ use thiserror::Error;
 
 use crate::{evaluator, lexer, parser};
 
-#[derive(Error, Debug)]
+/// The possible errors which can occur while parsing the string.
+#[derive(Error, Debug, PartialEq)]
 pub enum Error {
+    /// Symbolises that an error occured in the lexer.
     #[error(transparent)]
-    Lexer(#[from] lexer::Error),
+    Lexer(#[from] lexer::LexerError),
```

Comment:
Either `lexer::Error` or `LexerError` instead of `lexer::LexerError`

Reply:
fixed